app/utils/chart-config/versioning.ts

```diff
@@ -1454,6 +1454,44 @@ export const chartConfigMigrations: Migration[] = [
         delete newConfig.fields.color;
       }
 
+      return newConfig;
+    },
+  },
+  {
+    description: `ALL {
+      interactiveFiltersConfig {
+        legend {
+          + visible
+        }
+      }
+    }`,
+    from: "4.1.0",
+    to: "4.2.0",
+    up: (config) => {
+      const newConfig = {
+        ...config,
+        version: "4.2.0",
+      };
+
+      if (newConfig?.interactiveFiltersConfig?.legend) {
+        newConfig.interactiveFiltersConfig.legend = {
+          ...newConfig.interactiveFiltersConfig.legend,
+          visible: true,
+        };
+      }
+
+      return newConfig;
```

Comment:
@noahonyejese I think we shouldn't make the title visible by default for old chart, also see one of my previous comments 👍 

Btw. the `showTitle` parameter in chart config could be made optional with `t.partial`, so in fact we wouldn't need to have migrations at all, as old charts would still have it set to undefined, and for new we'd initialize the types correctly. But it's also fine to have it required, but then we should set it to false in migrations 👍